src/operator/nn/upsampling-inl.h

```diff
@@ -48,16 +48,18 @@ enum UpSamplingMultiInputMode {kConcat, kSum};
 }  // namespace up_enum
 
 struct UpSamplingParam : public dmlc::Parameter<UpSamplingParam> {
-  int scale;
+  TShape scale;
   int num_filter;
   int sample_type;
   int num_args;
   int multi_input_mode;
   uint64_t workspace;
   DMLC_DECLARE_PARAMETER(UpSamplingParam) {
     DMLC_DECLARE_FIELD(scale)
-    .set_range(1, 1000)
-    .describe("Up sampling scale");
+    .set_default(TShape())
```

Comment:
Is this going to break backward compatibility?

Reply:
I believe it is still backwards compatible, as it supports a scale as either an integer or a tuple.

Reply:
Can you use mxnet::TShape to clarify that you're not using nnvm::TShape or vice-versa.

Can you please explain how this will work for passing in a scalar argument?  I dont see TShape constructor taking a single scalar value.